### PR TITLE
Zachmu/window funcs

### PR DESCRIPTION
**Pull request by zachmu**

### CODE REVIEW

**Comment by zachmu** [comment]
@Hydrocharged this is ready to look at now

File: go/vt/sqlparser/sql.y [hunk 1]
```diff
@@ -2478,7 +2510,18 @@ as_ci_opt:
   }
 
 col_alias:
-  sql_id
+  ID
+    {
```
Comment (Hydrocharged) [inline go/vt/sqlparser/sql.y:2514]: Formatting here is a bit inconsistent with 4 spaces

File: go/vt/sqlparser/sql.y [hunk 2]
```diff
@@ -4141,6 +4353,37 @@ non_reserved_keyword:
 | YEAR
 | ZEROFILL
 
+// Reserved keywords that cause grammar conflicts in some places, but are safe to use as column name / alias identifiers.
```
Comment (Hydrocharged) [inline go/vt/sqlparser/sql.y:4356]: I'd add that these should also be declared in `reserved_keyword`.

**Review by Hydrocharged — APPROVED** [review]
LGTM! Just a few small things.

File: go/vt/sqlparser/sql.y [hunk 3]
```diff
@@ -4141,6 +4353,37 @@ non_reserved_keyword:
...
+| BIT_XOR
+| COUNT
+| CUME_DIST
+| DENSE_RANK
+| FIRST_VALUE
+| JSON_ARRAYAGG
+| JSON_OBJECTAGG
+| LAG
+| LAST_VALUE
+| LEAD openb
```
Comment (Hydrocharged) [inline go/vt/sqlparser/sql.y:4370]: Accidental `openb` here.

**Pull request merged**